### ISSUE TITLE
Mixed array fields are not being marked nullable

### ISSUE THREAD

**Issue by sureshjoshi** [apple/swift-openapi-generator#142]
I have an API call which is erroneously returning `null` inside of an array, so I've been trying to handle it in the OpenAPI yaml (I have no access to the API). 

```json
"stupidField": [
    null,
    { ... },
    null,
    ...
]
```

I believe this should work, where I allow for a second fallback value, which is a nullable string (or nullable anything).

```
stupidField:
    type: array
    items:
    oneOf:
        - $ref: "#/components/schemas/StupidFieldVm"
        - type: string
          nullable: true
    nullable: true
```

However, the generator removes the nullable trait from my String - so, I'm still getting crashes when trying to parse this JSON.

```
 @frozen public enum stupidFieldPayloadPayload: Codable, Equatable, Hashable, Sendable {
/// - Remark: Generated from `#/components/schemas/ClientVm/stupidFieldPayload/case1`.
case StupidFieldVm(Components.Schemas.StupidFieldVm)
/// - Remark: Generated from `#/components/schemas/ClientVm/stupidFieldPayload/case2`.
case case2(Swift.String)
/// Parsed a case that was not defined in the OpenAPI document.
case undocumented(OpenAPIRuntime.OpenAPIValueContainer)
public init(from decoder: any Decoder) throws {
    do {
        self = .StupidFieldVm(try .init(from: decoder))
        return
    } catch {}
    do {
        self = .case2(try .init(from: decoder))
        return
    } catch {}
    let container = try decoder.singleValueContainer()
    let value = try container.decode(OpenAPIRuntime.OpenAPIValueContainer.self)
    self = .undocumented(value)
}
```

I've tried alternate approaches, by allowing `AnyValue` or a few other options, but nothing appears to work.

Is there a workaround where I can conditionally ignore just that `stupidField` in my generator or decoder? 

I'm a bit stuck unfortunately, as I'm not sure how I can either pre-process the deeply nested JSON response and then decode it, or just somehow ignore the field, or conditionally custom decode just that field.

**Comment by czechboy0** [comment]
Hi @sureshjoshi – what you're encountering is a missing feature, the support for nullable schemas (#82). Now, the generator does support non-`required` properties in an object, and generates optional values for those.

I suggest you subscribe to updates on #82 to follow that feature being added.

For your specific case, you could try to build a ClientMiddleware where you parse the JSON body using JSONSerialization, remove/update that field, and serialize it back into data, making it look valid to the generated code. It's a workaround that won't have great performance, but it might be able to unblock you for now.\

ClientMiddleware docs: https://swiftpackageindex.com/apple/swift-openapi-runtime/0.1.5/documentation/openapiruntime/clientmiddleware

If this does resolve your issue, please close this issue as it's mostly a duplicate of #82 the way I understand it. Thanks! 🙏

**Comment by sureshjoshi** [comment]
Thanks for the quick reply! I was "worried" ClientMiddleware was the solution. It's gonna get sloppy in there :) Perf isn't a big deal in this case, as the stuff I have zero control over, I grab once per day - while the stuff I have control over, I can just ensure it's handled correctly locally.

Sorry for missing on #82 - I saw that ticket, but I didn't realize that it applied to my case with the `anyOf` workaround as well.

Will report back with my example workaround when complete - for those in the future.

**Comment by sureshjoshi** [comment]
Closing in favour of tracking ticket #82
